editor/src/clj/editor/app_view.clj

```diff
@@ -792,32 +935,22 @@
         (workspace/save-build-cache! workspace)
         build-results))))
 
-(defn- cached-build-target-output? [node-id label evaluation-context]
-  (and (= :build-targets label)
-       (project/project-resource-node? node-id evaluation-context)))
-
-(defn- update-system-cache-build-targets! [evaluation-context]
-  ;; To avoid cache churn, we only transfer the most important entries to the system cache.
-  (let [pruned-evaluation-context (g/pruned-evaluation-context evaluation-context cached-build-target-output?)]
-    (g/update-cache-from-evaluation-context! pruned-evaluation-context)))
-
 (defn- build-handler [project workspace prefs web-server build-errors-view main-stage tool-tab-pane]
   (let [project-directory (io/file (workspace/project-path workspace))
-        evaluation-context (g/make-evaluation-context)
         main-scene (.getScene ^Stage main-stage)
         render-build-error! (make-render-build-error main-scene tool-tab-pane build-errors-view)]
     (build-errors-view/clear-build-errors build-errors-view)
-    (async-build! project evaluation-context prefs {:debug? false} (workspace/artifact-map workspace)
+    (async-build! project prefs {:debug? false} (workspace/artifact-map workspace)
                   (make-render-task-progress :build)
                   (fn [build-results engine-descriptor build-engine-exception]
-                    (update-system-cache-build-targets! evaluation-context)
                     (when (handle-build-results! workspace render-build-error! build-results)
                       (when engine-descriptor
                         (show-console! main-scene tool-tab-pane)
                         (launch-built-project! project engine-descriptor project-directory prefs web-server false))
                       (when build-engine-exception
                         (log/warn :exception build-engine-exception)
-                        (engine-build-errors/handle-build-error! render-build-error! project evaluation-context build-engine-exception)))))))
+                        (g/with-auto-evaluation-context evaluation-context
+                          (engine-build-errors/handle-build-error! render-build-error! project evaluation-context build-engine-exception))))))))
```

Comment:
We need to create a fresh `evaluation-context` after the build completes in order to see the changes made by pre-/post-build hooks.